src/main/java/app/controllers/PresentationController.java

```diff
@@ -79,6 +73,22 @@ public String updateTimeSlots(@PathVariable Long pid,
                 return "redirect:/facultyMenu";
             }
         }
+        if(!action.toLowerCase().equals("update")) {
+            if(authenticatedUser.getType().equals("Student")) {
+                return "redirect:/project/" + pid + "/presentation";
+            }
+            proj.removeTimeSlot(new Long(action));
```

Comment:
Shouldn't action be `delete` and you pass in the timeSlot id via the request param?

Reply:
Id have to add a hidden field for the param which seems hacky